drivers/counter/counter_nrfx_timer.c

```diff
@@ -283,7 +283,14 @@ static uint32_t get_pending_int(const struct device *dev)
 static int init_timer(const struct device *dev,
 		      const struct counter_timer_config *config)
 {
-	const struct counter_nrfx_config *nrfx_config = dev->config;
+	struct counter_nrfx_config *nrfx_config = (struct counter_nrfx_config *) dev->config;
+
+#if defined(CONFIG_SOC_SERIES_BSIM_NRFXX)
+	/* For simulated devices we need to convert the hardcoded DT address from the real
+	 * peripheral into the correct one for simulation */
+	nrfx_config->timer = nhw_get_periph_base_addr(nrfx_config->timer);
```

Comment:
since timer can be now mutated, shouldn't it be placed in data?

Reply:
Updated as agreed on offline discussion